dali/kernels/common/block_setup.h

```diff
@@ -83,13 +89,25 @@ class BlockSetup {
     "Channel dimension must be in range [0..ndim] or -1 (no channel dim)");
   using BlockDesc = kernels::BlockDesc<ndim>;
 
-  BlockSetup() {
-    ivec<ndim> block_size(1);
+  template <int ndim>
+  using coord_vec = std::conditional_t<ndim == 1, i64vec<1>, ivec<ndim>>;
+
+  using coord_t = std::conditional_t<ndim == 1, int64_t, int32_t>;
```

Comment:
I'm a little bit lost here. Shouldn't `ndim` of `coord_vec` (passed to some function, e.g. `VariableBlockSize(coord_vec<n>&)`) be always the same as `BlockSetup::ndim`?

Reply:
Hmm, you're right, I can probably make it non-templated alias.